### ISSUE TITLE
Using 2fa app passwords in Gmail is confusing because it chooses OAUTH instead by default

### ISSUE THREAD

**Issue by alexlehm** [vert-x3/vertx-mail-client#145]
#### Describe the feature

Since Paulo implemented the XOAUTH feature in the mail client, using gmail and outlook requires twiddling with the supported auth methods list if XOAUTH is not used but the app-specific passwords (PLAIN or LOGIN), which is kind of confusing there were some issues mentioning that.
I wonder if it would be better to remove XOAUTH by default from the list of methods and only enable that when it is explicitly done with the config.
This is not so much a feature request, rather something I noticed when reading through emails from the issues.

If i remember correctly, XOATH also uses the user and password parameters like PLAIN, so its not possible to distinguish which is wanted when the programmer only sets the parameters with a method preference. I think a reasonable default would be good and then to mention which can be set when using the other method. The automatic select of the auth methods is kind of at a loss with this.

#### Contribution

I don't think I have time (maybe), otherwise @pmlopes would be the most obvious person because he did the OAUTH feature

**Comment by alexlehm** [comment]
actually issue #124 may fix that, however that would mean that the client would always try on failing oauth attempt first

**Comment by pmlopes** [comment]
Yes, that and allowing the user to set the order of the algorithms (to reduce the failure tests) should fix it.

**Comment by javadevmtl** [comment]
Question does the vertx-mail client now natively support XOAUTH or we need to do extra OAUTH first and then pass the tokens to the email client?

**Comment by gaol** [comment]
@javadevmtl you need to pass the tokens to the email client.
